tests/test-doc-portal.c

```diff
@@ -687,6 +687,55 @@ test_add_named (void)
   assert_doc_not_exist (id1, basename1, "com.test.App2");
 }
 
+static void
+test_get_host_paths (void)
+{
+  g_autofree char *doc_id = NULL;
+  g_autofree char *expected_real_path = NULL;
+  g_autofree char *real_path = NULL;
+  const char *basename = "host-path";
+  g_autoptr(GVariant) path= NULL;
+  g_autoptr(GVariant) reply = NULL;
+  g_autoptr(GVariant) result = NULL;
+  g_autoptr(GVariantIter) iter = NULL;
+  GVariantBuilder builder;
+  GError *error = NULL;
+  const gchar* key = NULL;
+
+  if (!check_fuse_or_skip_test ())
+    return;
+
+  doc_id = export_new_file (basename, "content", FALSE);
+
+  g_variant_builder_init (&builder, G_VARIANT_TYPE ("(as)"));
+  g_variant_builder_open (&builder, G_VARIANT_TYPE ("as"));
+  g_variant_builder_add (&builder, "s", doc_id);
+  g_variant_builder_close (&builder);
+
+  reply = g_dbus_connection_call_sync (session_bus,
+                                       "org.freedesktop.portal.Documents",
+                                       "/org/freedesktop/portal/documents",
+                                       "org.freedesktop.portal.Documents",
+                                       "GetHostPaths", g_variant_builder_end (&builder),
+                                       G_VARIANT_TYPE ("(a{say})"),
+                                       0, -1,
+                                       NULL,
+                                       &error);
+
+  g_assert_no_error (error);
+  result = g_variant_get_child_value (reply, 0);
+
+  g_assert (g_variant_is_of_type (result, G_VARIANT_TYPE ("a{say}")));
+
+  expected_real_path = g_build_filename (outdir, basename, NULL);
+  iter = g_variant_iter_new (result);
+  while (g_variant_iter_loop (iter, "{&s@ay}", &key, &path)) {
+    g_assert_cmpstr (key, ==, doc_id);
+    g_assert_cmpstr (g_variant_get_bytestring (path), ==, expected_real_path);
+    return;
+  }
```

Comment:
Please follow the project's code style:

```c
  while (g_variant_iter_loop (iter, "{&s@ay}", &key, &path))
    {
      ...
    }
```

Reply:
So I think you might have the opposite problem, no? Declaring key and path as `g_auto*` variables will end up double freeing these values I think?

Reply:
`key` is just `const gchar *`